### PR TITLE
Implement console_log for sapp-wasm

### PR DESCRIPTION
**Pull request by canadaduane**
Looking for your feedback here. Is this how you'd like to make `console_log` available?

### CODE REVIEW

File: src/console_log.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+/// Initializes the global logger setting `max_log_level` to the given value.
+pub fn init_with_level(level: Level) -> Result<(), SetLoggerError> {
+    log::set_logger(&LOGGER)?;
+    log::set_max_level(level.to_level_filter());
+    Ok(())
+}
+
+/// Initializes the global logger with `max_log_level` set to `Level::Info` (a sensible default).
+pub fn init() -> Result<(), SetLoggerError> {
+    init_with_level(Level::Info)
```
Comment (canadaduane) [inline src/console_log.rs]: Oops, will fix this

**Comment by not-fl3** [comment]
Wow, thanks! Thats really nice improvenet - logging on wasm is a big deal! 

My inputs: 

AFAIK rust libraries never embed logging backends. Thats the idea of `log-rs` - that user can choose logging backend implementation. So most idiomatic solution - make separate crate, `sapp-console-log` or something. And let users enable or not that crate in their code. 

But on the other hand - `miniquad` is trying to abstract over platforms and logging is part of that job. 
Filesystem, graphics, even random - are covered by `miniquad`, but logging will require platform dependent `#cfg(...)` and special dependencies for each platform in users code. 
For web - `sapp-console-log` will be actually the only possible logger. And same will be for android/ ios. 

In other words - should we deal with logging in miniquad at all? 

I think we should consider logging as a non-goal for miniquad and do this:

- make separate crate `sapp-console-logger`
- use `sapp-console-logger` in miniquad's user code - more higher level engines. Like in `good-web-game` or `macroquad`
- for `good-web-game` case - add `wasm_log_level` field into the `good_web_game::conf::Conf`. Let it unused on non-wasm platform, but use to choose log level for `sapp-console-logger`
- mention `env-logger` for PC and `sapp-console-logger` for WASM as recommended way to get logs from miniquad. 

Alternatives: make cross platform logging as a part of miniquad. Add feature gates to allow users to override choosen log backend, but make logging works on all platforms by default. The good thing - having `sapp-console-log` crate we can make this decision later.

**Comment by canadaduane** [comment]
What do you think of making other `console.*` log levels available as well? i.e. the way console_log [maps them here](https://github.com/iamcodemaker/console_log/blob/master/src/lib.rs#L121).

**Comment by canadaduane** [comment]
I'm trying to imagine what the code UI for the game developer would look like. Something like this?

```rs
#[macro_use] extern crate log;

#[cfg(target_arch = "wasm32")]
extern crate sapp_console_logger;

#[cfg(not(target_arch = "wasm32"))]
extern crate env_logger;

use log::Level;

fn main() {
    #[cfg(target_arch = "wasm32")]
    sapp_console_logger::init();

    #[cfg(not(target_arch = "wasm32"))]
    env_logger::init();

    debug!("this is a debug {}", "message");
    error!("this is printed by default");

    if log_enabled!(Level::Info) {
        let x = 3 * 4; // expensive computation
        info!("the answer was: {}", x);
    }
}
```

**Comment by not-fl3** [comment]
> I'm trying to imagine what the code UI for the game developer would look like. Something like this?

Yes, exactly. And with ~4 more lines for android/ios. 

I would like to put this code inside `macroquad`. But, actually, for `gwg` maybe it will make sense to keep it outside the library, like it was in [ggez](https://github.com/ggez/ggez/blob/master/examples/logging.rs). Not sure here to be honest.

**Comment by not-fl3** [comment]
> What do you think of making other console.* log levels available as well? i.e. the way console_log maps them here.

Yes, this would be nice! And good reason to split and rename poorly named `test_log` to - separate `console_log`, `console_warn`, `console_error`

**Pull request closed without merging**